ops/terraform/services/migrator/deploy.groovy

```diff
@@ -138,4 +156,32 @@ boolean canMigratorDeploymentProceed(String sqsQueueName, String awsRegion) {
     }
 }
 
+// checks to determine whether the migrator deployment is required
+// returns true when the latest versioned migration script is newer
+// than the database's stored schema version
+// otherwise false
+boolean isMigratorDeploymentRequired(String bfdEnv, String awsRegion) {
+    println "Comparing schema migration versions..."
+    // Initialize stored schema version
+    int storedSchemaVersion = 0
+    // check SSM Parameter Store
+    try {
+        storedSchemaVersion = awsSsm.getParameter(
+                parameterName: "/bfd/${bfdEnv}/common/nonsensitive/database_schema_version",
+                awsRegion: awsRegion
+        ) as Integer
+    } catch(groovy.lang.MissingPropertyException ex){
```

Comment:
I don't believe this will function as expected with`groovy.lang.MissingPropertyException` being a little too specific. I wrote up a tiny test pipeline to evaluate sh returns because I was curious, and as you might recall from our testing yesterday on `bfd-2200`, we get helpful hints from the awscli output (`An error occurred (ParameterNotFound) when calling the GetParameter operation:`), but a _somewhat_ less helpful error from Groovy/Jenkins (`ERROR: script returned exit code 254`).

We probably ought to truncate this a bit to the following:

```suggestion
    } catch(Exception ex) {
```
Keen observers might note that this isn't handling other exceptions, like those that result from authentication failures, but... I think it's pretty obvious that `awsSsm.groovy` is an implementation that's _fairly_ specific to this use-case, which I think is fine.

Reply:
ok I thought you wanted to catch the specific exception instead of just the general exception.. Sorry for misunderstanding..